go/interpreter/lfvm/interpreter_intergration_test.go

```diff
@@ -0,0 +1,689 @@
+	tests = addTrivialTest(tests, gas(12), stackSize(2, 1), DUP2_MSTORE)
+	tests = addTrivialTest(tests, gas(14), stackSize(5, 3), AND_SWAP1_POP_SWAP2_SWAP1)
+
+	testWithData := func(gas gas, stack stackTest, requiredDataOps int, opCodes ...OpCode) []testDeclaration {
+		tests := make([]testDeclaration, len(opCodes))
+		for i, op := range opCodes {
+			code := attachDataToOp(op, requiredDataOps)
+			tests[i] = testDeclaration{
+				code:  code,
+				stack: stack,
+				gas:   cost(gas),
+			}
+		}
+		return tests
+	}
+
+	tests = append(tests, testWithData(15, stackSize(0, 1), 1, PUSH1_PUSH1_PUSH1_SHL_SUB)...)
+	tests = append(tests, testWithData(9, stackSize(1, 4), 2, PUSH1_PUSH4_DUP3)...)
+
+	///////////////////////////////////
+	// Jump instructions
+
+	tests = append(tests, []testDeclaration{
+
+		{
+			nameOverride: "JUMP",
+			code:         []Instruction{{PUSH1, 2 << 8}, {JUMP, 0}, {JUMPDEST, 0}},
+			gas:          cost(3 + 8 + 1),
+		},
+		{
+			nameOverride: "JUMPI",
+			code:         []Instruction{{PUSH1, 1 << 8}, {PUSH1, 3 << 8}, {JUMPI, 0}, {JUMPDEST, 0}},
+			gas:          cost(3 + 3 + 10 + 1),
+		},
+		{
+			nameOverride: "POP_JUMP",
+			code:         []Instruction{{PUSH1, 3 << 8}, {PUSH1, 0}, {POP_JUMP, 0}, {JUMPDEST, 0}},
+			gas:          cost(3 + 3 + 2 + 8 + 1),
+		},
+		{
+			nameOverride: "ISZERO_PUSH2_JUMPI",
+			code:         []Instruction{{PUSH1, 2 << 8}, {ISZERO_PUSH2_JUMPI, 0}, {JUMPDEST, 0}},
+			gas:          cost(3 + 3 + 3 + 10 + 1),
+		},
+		{
+			nameOverride: "PUSH2_JUMP",
+			code:         []Instruction{{PUSH2_JUMP, 1}, {JUMPDEST, 0}},
+			gas:          cost(3 + 8 + 1),
+		},
+		{
+			nameOverride: "PUSH2_JUMPI",
+			code:         []Instruction{{PUSH1, 2 << 8}, {PUSH2_JUMPI, 2}, {JUMPDEST, 0}},
+			gas:          cost(3 + 3 + 10 + 1),
+		},
+		{
+			nameOverride: "SWAP2_SWAP1_POP_JUMP",
+			code: []Instruction{
+				{PUSH1, 4 << 8},
+				{PUSH1, 0},
+				{PUSH1, 0},
+				{SWAP2_SWAP1_POP_JUMP, 0}, {JUMPDEST, 0}},
+			stack: stackSize(1, 2),
+			gas:   cost(3*3 + 3*2 + 2 + 8 + 1),
+		},
+		// LFVM jump_to extension
+		{
+			code: []Instruction{{JUMP_TO, 2}, {NOOP, 0}},
+		},
+	}...)
+
+	///////////////////////////////////
+	// Log
+
+	tests = append(tests,
+		generateTestCaseFor(func(op OpCode) testDeclaration {
+			n := int(op - LOG0)
+
+			stackBefore := make([]tosca.Word, n+2)
+			for i := 0; i < n+2; i++ {
+				stackBefore[i] = toWord(1) //< value 1
+			}
+
+			return testDeclaration{
+				code: []Instruction{{op, 0}},
+				mockCalls: func(mock *tosca.MockRunContext) {
+					mock.EXPECT().EmitLog(gomock.Any())
+
+				},
+				stack: stackWithValues(stackBefore, 0),
+				gas:   cost(gas(375 + (n * 375) + 8*1 + 3)),
+			}
+		}, getOpCodesInRange(LOG0, LOG4))...,
+	)
+
+	///////////////////////////////////
+	// TRIVIAL TESTS introduced in revisions
+
+	tests = append(tests, testDeclaration{
+		code:               []Instruction{{BASEFEE, 0}, {STOP, 0}},
```

Comment:
what is different between this `STOP` and the one introduced with "Special Opcodes" ?